### DISCUSSION TITLE
Issue accessing data using fastf1 api

### DISCUSSION THREAD

**Discussion by laurence9899** [theOehrly/Fast-F1#505] [Q&A] [answered]
Hi I'm trying to use the fastf1 api to pull data for a whole season. I created the below code with a data frame of the parameters needed for a the fastf1.api.make_path function. I am using a for loop to move through the entire data frame. However I keep having an issue with the code in which I get the below error. Anyone any ideas what I am doing wrong?

_Bahrain Grand Prix
2023-03-03
2023-03-05
req            INFO     No cached data found for _extended_timing_data. Loading data...
_api           INFO     Fetching timing data...
Traceback (most recent call last):
  File "/Users/laurencehearne/Library/CloudStorage/OneDrive-UniversityofLimerick/FYP/VSCode/scripts/aquire_timingdata.py", line 77, in <module>
    [laps_data,stream_data] = fastf1.api.timing_data(path)
  File "/Users/laurencehearne/anaconda3/envs/Scripts/lib/python3.10/site-packages/fastf1/_api.py", line 160, in timing_data
    = _extended_timing_data(path, response=response, livedata=livedata)
  File "/Users/laurencehearne/anaconda3/envs/Scripts/lib/python3.10/site-packages/fastf1/req.py", line 433, in _cached_api_request
    data = func(api_path, **func_kwargs)
  File "/Users/laurencehearne/anaconda3/envs/Scripts/lib/python3.10/site-packages/fastf1/_api.py", line 178, in _extended_timing_data
    raise SessionNotAvailableError(
fastf1._api.SessionNotAvailableError: No data for this session! If this session only finished recently, please try again in a few minutes._

```
`data = {
    'GrandPrix': [
        'Bahrain Grand Prix', 'Saudi Arabia Grand Prix', 'Australian Grand Prix', 
        'Azerbaijan Grand Prix', 'Miami Grand Prix', 'Monaco Grand Prix', 
        'Spanish Grand Prix', 'Canadian Grand Prix', 'Austrian Grand Prix', 
        'British Grand Prix', 'Hungarian Grand Prix', 'Belgian Grand Prix', 
        'Dutch Grand Prix', 'Italian Grand Prix', 'Singapore Grand Prix', 
        'Japanese Grand Prix', 'Qatar Grand Prix', 'United States Grand Prix', 
        'Mexico City Grand Prix', 'São Paulo Grand Prix', 'Las Vegas Grand Prix',
        'Abu Dhabi Grand Prix'
    ],
    'StartDate': [
        '2023-03-03', '2023-03-17', '2023-03-31', 
        '2023-04-28', '2023-05-05', '2023-05-26', 
        '2023-06-02', '2023-06-16', '2023-06-30', 
        '2023-07-07', '2023-07-21', '2023-07-28', 
        '2023-08-25', '2023-09-01', '2023-09-15', 
        '2023-09-22', '2023-10-06', '2023-10-20', 
        '2023-10-27', '2023-11-03', '2023-11-16', 
        '2023-11-24'
    ],
    'Session': ['Race'] * 22,
    'RaceDate': [
        '2023-03-05', '2023-03-19', '2023-04-02', 
        '2023-04-30', '2023-05-07', '2023-05-28', 
        '2023-06-04', '2023-06-18', '2023-07-02', 
        '2023-07-09', '2023-07-23', '2023-07-30', 
        '2023-08-27', '2023-09-03', '2023-09-17', 
        '2023-09-24', '2023-10-08', '2023-10-22', 
        '2023-10-29', '2023-11-05', '2023-11-18', 
        '2023-11-26'
    ]
}
f1_2023_season_df = pd.DataFrame(data)

year=2023
race=1
# Loop through each row in the DataFrame
for index, row in f1_2023_season_df.iterrows():
    
    grand_prix = row['GrandPrix']
    start_date = row['StartDate']
    session = row['Session']
    race_date = row['RaceDate']
    f1_2023_season_df = pd.DataFrame(data)
    print(grand_prix)
    print(start_date)
    print(race_date)
    path=fastf1.api.make_path(grand_prix, start_date, 'Race', race_date)

    [laps_data,stream_data] = fastf1.api.timing_data(path)
`
```

**Comment by Casper-Guo** [comment, marked as answer]
Is there a reason why you are not using `get_session()` instead? It also saves you the trouble of hard coding the race names and start dates.

From the docs:

> [fastf1.api](https://docs.fastf1.dev/api.html#module-fastf1.api) will be considered private in future releases and potentially be removed or changed. Please do not use functionality from [fastf1.api](https://docs.fastf1.dev/api.html#module-fastf1.api). If you currently require functionality from there, please open an issue on Github with details about what you require and why.

**Comment by laurence9899** [comment]
No, no reason why not. I am a newcomer to fastf1, apologies for that.  I have changed it to that now and it's working. Thank you very much.
